File: 백준/Gold/1759. 암호 만들기/암호 만들기.py
import sys
from itertools import combinations

L, C = map(int,sys.stdin.readline().split())
arr = list(sys.stdin.readline().split())
combi = list(combinations(arr,L))
answer = []
for com in combi :
    flag = False
    flag2 = False
    cnt = 0
    for c in com :
        if c in ['a','e','i','o','u'] :
            flag = True
        elif c not in ['a', 'e', 'i', 'o', 'u']:
            cnt += 1
            if cnt >= 2:
                flag2 = True
    if flag == True and flag2 == True :
        temp = list(com)
        temp.sort()
        answer.append(''.join(temp))
answer.sort()
for ans in answer :
    print(ans)

File: 백준/Gold/1759. 암호 만들기/암호 만들기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        int L = Integer.parseInt(input[0]);
        int C = Integer.parseInt(input[1]);

        String[] arr = br.readLine().split(" ");
        Arrays.sort(arr); // Sort the input array

        List<String> combi = new ArrayList<>();
        generateCombinations(arr, L, 0, "", combi);

        List<String> answer = new ArrayList<>();
        for (String com : combi) {
            int vowelCount = 0;
            for (char c : com.toCharArray()) {
                if (isVowel(c)) {
                    vowelCount++;
                }
            }
            if (vowelCount >= 1 && L - vowelCount >= 2) {
                answer.add(com);
            }
        }

        for (String ans : answer) {
            System.out.println(ans);
        }
    }

    private static void generateCombinations(String[] arr, int length, int index, String current, List<String> result) {
        if (length == 0) {
            result.add(current);
            return;
        }
        if (index >= arr.length) {
            return;
        }
        generateCombinations(arr, length - 1, index + 1, current + arr[index], result);
        generateCombinations(arr, length, index + 1, current, result);
    }

    private static boolean isVowel(char c) {
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
    }
}